admin/src/main/java/com/exadel/frs/scheduler/job/StatisticsJob.java

```diff
@@ -0,0 +1,93 @@
+package com.exadel.frs.scheduler.job;
+
+import com.exadel.frs.entity.Face;
+import com.exadel.frs.entity.Model;
+import com.exadel.frs.exception.ApperyServiceException;
+import com.exadel.frs.repository.FacesRepository;
+import com.exadel.frs.repository.InstallInfoRepository;
+import com.exadel.frs.repository.ModelRepository;
+import com.exadel.frs.system.feign.ApperyStatisticsClient;
+import com.exadel.frs.system.feign.StatisticsFacesEntity;
+import feign.FeignException;
+import lombok.NoArgsConstructor;
+import org.apache.commons.lang3.Range;
+import org.quartz.JobExecutionContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.quartz.QuartzJobBean;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang3.Range.between;
+
+@NoArgsConstructor
+@Component
+public class StatisticsJob extends QuartzJobBean {
+
+    @Value("${app.feign.appery-io.api-key}")
+    private String statisticsApiKey;
+    private ApperyStatisticsClient apperyStatisticsClient;
+    private InstallInfoRepository installInfoRepository;
+    private FacesRepository facesRepository;
+    private ModelRepository modelRepository;
+
+    private List<Range> ranges = List.of(
+                Range.between(1, 10),
+                Range.between(11, 50),
+                Range.between(51, 200),
+                Range.between(201, 500),
+                Range.between(501, 2000),
+                between(2001, 10000),
+                between(10001, 50000),
+                between(50001, 200000),
+                between(200001, 1000000)
+    );
+
+    @Autowired
+    public StatisticsJob(final ApperyStatisticsClient apperyStatisticsClient, final InstallInfoRepository installInfoRepository, final FacesRepository facesRepository, final ModelRepository modelRepository) {
+        this.apperyStatisticsClient = apperyStatisticsClient;
+        this.installInfoRepository = installInfoRepository;
+        this.facesRepository = facesRepository;
+        this.modelRepository = modelRepository;
+    }
+
+    @Override
+    public void executeInternal(final JobExecutionContext context) {
+        List<Model> models = modelRepository.findAll();
+        Map<String, String> facesModelGuidAndRangeMap = new HashMap<>();
+
+        for (Model model : models) {
+            String apiKey = model.getApiKey();
+            List<Face> faces = facesRepository.findByApiKey(apiKey);
```

Comment:
Can we replace `modelRepository.findAll()` and `facesRepository.findByApiKey(apiKey)` with a single SQL request like 
`SELECT * FROM Faces WHERE f.apiKey in (SELECT apiKey FROM models)` ?